server/api.js

```diff
@@ -2,100 +2,416 @@ import { Router } from "express";
+			} else {
+				const query = "UPDATE plans SET description=$2 WHERE id=$1 RETURNING *";
+				pool
+					.query(query, [plan_id, description])
+					.then(() =>
+						res.json({ success: "plan has been Updated!", user: result.rows })
+					)
+					.catch((err) => console.error(err.message));
+			}
+		});
+});
+
+// Delete a plan form specific user with plan id <==== have a problem
+router.delete("users/:user_id/plans/:plan_id", async (req, res) => {
+	const { user_id, plan_id } = req.params;
+
+	if (!user_id || !plan_id) {
+		res
+			.status(404)
+			.json({ status: 404, msg: "We couldn't find a plan with this id!" });
+	}
+
 	try {
-		console.log(req.body);
-		const { first_name, last_name, region, role, email, password } = req.body;
-		const newUser = await pool.query(
-			"INSERT INTO users (first_name, last_name, region, role, email, password ) VALUES ($1, $2, $3, $4, $5, $6) RETURNING *",
-			[first_name, last_name, region, role, email, password]
+		const deleteGoal = await pool.query("DELETE FROM goals WHERE plan_id=$1", [
+			plan_id,
+		]);
+
+		const deleteFeedback = await pool.query(
+			"DELETE FROM feedbacks WHERE plan_id=$1",
+			[plan_id]
+		);
+
+		const result = await pool.query(
+			"DELETE FROM plans WHERE plans.id=$1 AND user_id=$2",
+			[plan_id, user_id]
+		);
+
+		res.json({ Success: "Plan has been deleted" });
+	} catch (err) {
+		res.status(500).send("server error");
+	}
+});
+
+// GET goals form specific user id
+router.get("/users/:user_id/goals", (req, res) => {
+	const { user_id } = req.params;
+	const query = `SELECT p.id, p.user_id, p.description,ARRAY_AGG(g.title  || ' status:'|| g.status) goals_list
+					FROM goals g INNER JOIN plans p on p.id=g.plan_id
+					WHERE p.user_id=$1 GROUP BY p.id, p.user_id, p.description`;
+
+	pool
+		.query(query, [user_id])
+		.then((result) => {
+			if (result.rowCount) {
+				res.json(result.rows);
+			} else {
+				res.status(404).json({
+					status: 404,
+					msg: "User doesn't have any goals.",
+				});
+			}
+		})
+		.catch((err) => console.error(err.message));
+});
+
+// GET a goal form specific plan id
+router.get("/users/:user_id/plans/:plan_id/goals", (req, res) => {
+	const { user_id, plan_id } = req.params;
+	const query = `SELECT p.user_id, p.id plan_id, g.id goal_id,  g.title, g.status, g.start_date, g.end_date, g.create_date  FROM goals g 
+						INNER JOIN plans p on p.id = g.plan_id where p.user_id = $1 and g.plan_id=$2`;
+	pool
+		.query(query, [user_id, plan_id])
+		.then((result) => {
+			if (result.rowCount) {
+				res.json(result.rows);
+			} else {
+				res.status(404).json({
+					status: 404,
+					msg: "This user doesn't have a goal with this id",
+				});
+			}
+		})
+		.catch((err) => console.error(err.message));
+});
+
+//POST a goal for specific plan id
+router.post("/users/plans/:plan_id/goals", (req, res) => {
+	const { plan_id } = req.params;
+	const { title, status, start_date, end_date } = req.body;
```

Comment:
What if they are empty strings?, or if the status is an unsupported value?
We probably need some validation here

(Hint: you will need to define a list of supported values and check if the string is in that list -  same for user roles. To make your life easier define it as constant only **once**, so that it's easy to maintain).